src/parser.cpp

```diff
@@ -479,6 +479,104 @@ options_metadata parser::load_settings()
+        "The difficulty retargeting period in seconds, defaults to 1209600."
+    )
+    (
+        "bitcoin.proof_of_work_limit",
+        value<uint32_t>(&configured.bitcoin.proof_of_work_limit),
+        "The proof of work limit, defaults to 486604799."
+    )
+    (
+        "bitcoin.genesis_block",
+        value<config::block>(&configured.bitcoin.genesis_block),
+        "The genesis block."
+    )
+    (
+        "bitcoin.activation_threshold",
+        value<size_t>(&configured.bitcoin.activation_threshold),
+        "The number of new version blocks required for bip34 style soft fork activation, defaults to 750."
+    )
+    (
+        "bitcoin.enforcement_threshold",
+        value<size_t>(&configured.bitcoin.enforcement_threshold),
+        "The number of new version blocks required for bip34 style soft fork enforcement, defaults to 950."
+    )
+    (
+        "bitcoin.activation_sample",
+        value<size_t>(&configured.bitcoin.activation_sample),
+        "The number of blocks considered for bip34 style soft fork activation, defaults to 1000."
+    )
+    (
+        "bitcoin.bip65_freeze",
+        value<size_t>(&configured.bitcoin.bip65_freeze),
+        "The block height to freeze the bip65 softfork as in bip90, defaults to 388381."
+    )
+    (
+        "bitcoin.bip66_freeze",
+        value<size_t>(&configured.bitcoin.bip66_freeze),
+        "The block height to freeze the bip66 softfork as in bip90, defaults to 363725."
+    )
+    (
+        "bitcoin.bip34_freeze",
+        value<size_t>(&configured.bitcoin.bip34_freeze),
+        "The block height to freeze the bip34 softfork as in bip90, defaults to 227931."
+    )
+    (
+        "bitcoin.bip16_activation_time",
+        value<uint32_t>(&configured.bitcoin.bip16_activation_time),
+        "The activation time for bip16 in unix time, defaults to 1333238400."
+    )
+    (
+        "bitcoin.bip34_active_checkpoint",
+        value<config::checkpoint>(&configured.bitcoin.bip34_active_checkpoint),
+        "The hash:height checkpoint for bip34 activation, defaults to 000000000000024b89b42a942fe0d9fea3bb44ab7bd1b19115dd6a759c0808b8:227931."
+    )
+    (
+        "bitcoin.bip9_bit0_active_checkpoint",
+        value<config::checkpoint>(&configured.bitcoin.bip9_bit0_active_checkpoint),
+        "The hash:height checkpoint for bip9 bit0 activation, defaults to 000000000000000004a1b34462cb8aeebd5799177f7a29cf28f2d1961716b5b5:419328."
+    )
+    (
+        "bitcoin.bip9_bit1_active_checkpoint",
+        value<config::checkpoint>(&configured.bitcoin.bip9_bit1_active_checkpoint),
+        "The hash:height checkpoint for bip9 bit1 activation, defaults to 0000000000000000001c8018d9cb3b742ef25114f27563e3fc4a1902167f9893:481824."
+    )
+    (
+        "bitcoin.initial_block_subsidy_bitcoin",
+        value<uint64_t>()->notifier([&](uint64_t value)
+        {
+            configured.bitcoin.initial_block_subsidy_bitcoin(value);
+        }),
```

Comment:
How about further rationalization:

```cpp
    (
        "bitcoin.initial_block_subsidy_bitcoin",
        PROPERTY(uint64_t, configured.bitcoin.initial_block_subsidy_bitcoin),
        "The initial block subsidy in bitcoin, defaults to 50."
    )
```

Reply:
I'm not a fan of macros. They're not type safe and are a legacy carried over from C where they're actually a necessary evil. Modern C++ can do just fine without macros.

Reply:
Nobody is s fan of macros, but they serve a useful purpose in improving code readability and limiting repetition.

Reply:
Done. I've used a C++ template instead of a macro ;)